Improve Heartcore Management API documentation (#4902)
umbraco/UmbracoDocs: umbraco-heartcore/tutorials/creating-content-with-media.md

### Clean up the properties and use it as a template for your request body

Finally, we need to clean up the properties, so it can be used as a template for our request body.

In this case, we can omit ID's and create/update dates as these are auto-generated at creation time:

```json
{
//"_currentVersionState": {
//  "$invariant": "PUBLISHED"
//},
  "name": {
    "$invariant": "Star Wars Blog"
  },
//"_updateDate": {
//  "$invariant": "2023-02-08T13:26:16.6Z"
//},
//"_hasChildren": false,
//"_level": 1,
//"_createDate": "2023-02-08T13:21:35.27Z",
//"_id": "a6126eb4-c667-466c-ae7c-252a5b2e7be9",
  "contentTypeAlias": "blog",
//"sortOrder": 0,
  "image": {
    "$invariant": [{
      //"key": "394c9bfd-be7b-486d-968a-ce844aa76b5e",
        "mediaKey": "766c9c5b-f4eb-4a5c-b94e-06701eafcbab",
      //"crops": [],
      //"focalPoint": null
      }]
  }
}
```

Resulting in the following request body:

```json
//You cannot use this request body as-is, as the mediaKey will differ.
{
  "name": {
    "$invariant": "Star Wars Blog"
  },
  "contentTypeAlias": "blog",
  "image": {
    "$invariant": [{
        "mediaKey": "766c9c5b-f4eb-4a5c-b94e-06701eafcbab",
      }]
  }
}
```

> The media key is the ID of the image in the media section, and as such it is the image reference to the image the Media Picker V3 property uses.

Reviewer: ```suggestion
The media key is the ID of the image in the media section. As such it is the image reference to the image the Media Picker V3 property uses.
```

See the [Media Picker 3](../../umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/media-picker-3.md) section for more info on the other properties.

## Creating a Media Picker with the Management API

> After having followed the steps demonstrated in the [previous section](#determining-the-request-body), we can now create a new content node with a Media Picker using the Management API.

Reviewer: ```suggestion 
Following the steps demonstrated in the [previous section](#determining-the-request-body), we can now create a new content node with a Media Picker using the Management API.
```

    ![Opening the non-GET request modal](images/management-api-browser-opening-the-non-get-requests-modal.png)

3. Next copy and paste your request body into the request body section in the modal.

    ![Making a non-GET request](images/management-api-browser-non-get-request.png)

4. Click the `Make Request` button.

Now you have created a new content node with a Media Picker property editor that correctly references the image with the ID "766c9c5b-f4eb-4a5c-b94e-06701eafcbab".

## Publishing a Media Picker with the Management API

After having created the content node it is time to publish it, so it can be viewed on the front-end.

To publish a content node with a Media Picker, follow these steps:

1. Go to the Management API Browser, fill in the API Key as a Custom Request Header, Write the request URL for the content, and click the `Go!` button.
2. Navigate to and click the small orange exclamation mark button next to the `publish` link in the `Links` section.
3. Change the request type to `PUT`, and leave the request body empty as an empty object.
4. Click the `Make Request` button.

    ![Publishing a content node](images/management-api-browser-publishing-a-content-node.png)

> And you are done! You should now be able to upload a media item and create a new content node that uses the newly uploaded media item in a Media Picker v3 property.

Reviewer: ```suggestion
And you are done! You are now able to upload a media item and create a new content node using the newly uploaded media item in a Media Picker v3 property.
```

> You can use a similar approach for discovering the format of more complex content for when you want to create it through the Content Management API.

Reviewer: ```suggestion
You can use a similar approach for discovering the format of more complex content when you want to create it through the Content Management API.
```

> Please note that the .NET Client Library also supports the use of the Management API if you want to do it from C# code.

Reviewer: I think instead of writing "please note" it would make sense to wrap this sentence in a tip and then change the sentence a bit to something like:

{% hint style="info" %}
If you want to do it from C# code, the .NET client library also supports the use of the Management API.
{% endhint %}

Let me know if it makes sense 😄

Author: It does. I will make the change 👍🏻

//},
  "name": {
    "$invariant": "Star Wars Blog"
  },
//"_updateDate": {
//  "$invariant": "2023-02-08T13:26:16.6Z"
//},
//"_hasChildren": false,
//"_level": 1,
//"_createDate": "2023-02-08T13:21:35.27Z",
//"_id": "a6126eb4-c667-466c-ae7c-252a5b2e7be9",
  "contentTypeAlias": "blog",
//"sortOrder": 0,
  "image": {
    "$invariant": [{
      //"key": "394c9bfd-be7b-486d-968a-ce844aa76b5e",
        "mediaKey": "766c9c5b-f4eb-4a5c-b94e-06701eafcbab",
      //"crops": [],
      //"focalPoint": null
      }]
  }
}
```

Resulting in the following request body:

```json
//You cannot use this request body as-is, as the mediaKey will differ.
{
  "name": {
    "$invariant": "Star Wars Blog"
  },
  "contentTypeAlias": "blog",
  "image": {
    "$invariant": [{
        "mediaKey": "766c9c5b-f4eb-4a5c-b94e-06701eafcbab",
      }]
  }
}
```

The media key is the ID of the image in the media section. As such it is the image reference to the image the Media Picker V3 property uses.

See the [Media Picker 3](../../umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/media-picker-3.md) section for more info on the other properties.

{% hint style="info" %}

> If you decide to copy the json from the response section to another editor/application, be careful that the formatting is correct, and that no extra characters are added. These characters can be hidden, but will cause the request to fail.

Reviewer: ```suggestion
If you copy the JSON from the response section to another editor/application make sure the formatting is correct, and no extra characters are added. These characters can be hidden but will cause the request to fail.
```
